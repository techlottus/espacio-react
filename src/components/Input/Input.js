import React, { useEffect } from 'react'

export const Input = React.memo(({data,value,hasError,errorMessage ,eventFocus,eventKeyPress, eventBlurPress, listenIcon})  => {
    const inputRef = React.createRef();

    useEffect(() => {
        inputRef.current.info = {
            label: data.label || '',
            name: data.name || '',
            type: data.type || '',
            typeButton: data.typeButton || '',
            maxlength: data.maxlength || '',
            onPaste: data.onPaste,
            placeholder: data.placeholder || '',
            autocomplete: data.autocomplete || '',
            disabled: data.disabled,
            alphanumeric: data.alphanumeric,
            alphabetical: data.alphabetical,
            onlyNumbers: data.onlyNumbers,
            upperCase: data.upperCase,
            pattern: data.pattern || '',
            mask: data.mask || ""
        }
    },[data]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        inputRef.current.value = value;
    },[value]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        inputRef.current.hasError = hasError;
        inputRef.current.errorMessage = errorMessage;
    },[hasError,errorMessage]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {

        inputRef.current.addEventListener('eventFocus', eventFocus);
        inputRef.current.addEventListener('eventKeyPress', eventKeyPress);
        inputRef.current.addEventListener('eventBlurPress', eventBlurPress);
        inputRef.current.addEventListener('listenIcon', listenIcon);
    }, []); // eslint-disable-line react-hooks/exhaustive-deps

    return(
        <>
        <lottus-input ref={inputRef}></lottus-input>
        </>
    )
})