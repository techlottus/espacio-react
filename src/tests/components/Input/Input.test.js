import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Input } from "../../../components/Input/Input";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState;
let data = {
    label: "",
    name: "",
    type: "",
    typeButton: "",
    maxlength: "",
    onPaste: true,
    placeholder: "",
    autocomplete: "",
    disabled: false,
    alphanumeric: false,
    alphabetical: true,
    onlyNumbers: false,
    upperCase: false,
    pattern: "",
}

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Input data={data} />
    </Provider>
  );
};

describe("Test <Input/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
