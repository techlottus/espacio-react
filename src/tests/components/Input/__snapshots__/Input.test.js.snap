// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Input/> should init component 1`] = `<lottus-input />`;
